Source/Events.Store.MongoDB/Processing/Streams/StreamProcessorStates.cs

```
@@ -27,60 +29,125 @@ public class StreamProcessorStates : IStreamProcessorStateRepository
-                // var persistedState = await states.Find(
-                //     _streamProcessorFilter.Eq(_ => _.ScopeId, streamProcessorId.ScopeId.Value)
-                //         & _streamProcessorFilter.Eq(_ => _.EventProcessorId, streamProcessorId.EventProcessorId.Value)
-                //         & _streamProcessorFilter.Eq(_ => _.SourceStreamId, streamProcessorId.SourceStreamId.Value))
-                //     .FirstOrDefaultAsync(cancellationToken)
-                //     .ConfigureAwait(false);
-                // if (persistedState != null)
-                // {
-                //     return (true, persistedState.ToRuntimeRepresentation());
-                // }
-                // else
-                // {
-                //     return (false, null);
-                // }
+                if (id is SubscriptionId subscriptionId)
+                {
+                    var states = await _connection.GetSubscriptionStateCollection(subscriptionId.ScopeId, cancellationToken).ConfigureAwait(false);
+                    var persistedState = await states.Find(CreateFilter(subscriptionId))
+                        .FirstOrDefaultAsync(cancellationToken)
+                        .ConfigureAwait(false);
+                    return (persistedState != null) ? (true, persistedState.ToRuntimeRepresentation()) : (false, null);
+                }
+                else
+                {
+                    var streamProcessorId = id as StreamProcessorId;
+                    var states = await _connection.GetStreamProcessorStateCollection(streamProcessorId.ScopeId, cancellationToken).ConfigureAwait(false);
+                    var persistedState = await states.Find(CreateFilter(streamProcessorId))
+                        .FirstOrDefaultAsync(cancellationToken)
+                        .ConfigureAwait(false);
+                    return (persistedState != null) ? (true, persistedState.ToRuntimeRepresentation()) : (false, null);
+                }
```

I think we should throw an exception if id is no StreamPRocessorId or SubscriptionId